### PR TITLE
Adds TCAF Equipment and Armor

### PR DESCRIPTION
**Pull request by NobleRow**
Creates a selection of Tau Ceti Armed Forces armor and equipment compatible with the modular armor system, including but not limited to, a uniform, gloves, armor, helmets, and pouches.

Most of the armor is heavily inspired by existing TCFL assets, with the chest plate in particular being a meld between several online references and the marine chestplate from the HaloStation server. The chestplate has been modified significantly from its original form to be fitting in line with the server's "No Pop Culture" rule and to make the armor more original.

I will be including the original chestplate in the asset licenses section despite the changes, to credit the original creator as I was learning how to sprite.

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| halostation/code/modules/halo/clothing/marine.dmi | HaloStation | CC0 |

### CODE REVIEW

**Comment by NobleRow** [comment]
!review

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -263,6 +263,13 @@
 	item_state = "legion_bag"
 	empty_delay = 0.8 SECOND
 
+/obj/item/storage/backpack/tcaf
+	name = "TCAF Carapace Backpack"
```
Comment (Generalcamo) [inline code/game/objects/items/weapons/storage/backpack.dm:267]:
```suggestion
	name = "\improper TCAF Carapace Backpack"
```

File: code/modules/clothing/head/berets.dm [hunk 2]
```diff
@@ -172,7 +172,7 @@
 	icon_state = "orion"
 	item_state = "orion"
 
-// TCFL
+// TCAF
 
 /obj/item/clothing/head/beret/legion
 	name = "TCFL dress beret"
```
Comment (Generalcamo) [inline code/modules/clothing/head/berets.dm:178]:
```suggestion
	name = "\improper TCFL dress beret"
```
Existing issue but we may as well tackle that now

File: code/modules/clothing/head/berets.dm [hunk 3]
```diff
@@ -192,6 +192,20 @@
 	icon_state = "tcfl_sentinel"
 	item_state = "tcfl_sentinel"
 
+/obj/item/clothing/head/beret/legion/tcaf
+	name = "TCAF dress beret"
```
Comment (Generalcamo) [inline code/modules/clothing/head/berets.dm:196]:
```suggestion
	name = "\improper TCAF dress beret"
```

File: code/modules/clothing/head/berets.dm [hunk 4]
```diff
@@ -192,6 +192,20 @@
...
+	desc = "A blue dress beret bearing the flag of the Republic of Biesel. Often only seen worn by in-service members of the Tau Ceti Armed Forces."
+	icon = 'icons/clothing/under/uniforms/tcaf_uniform.dmi'
+	contained_sprite = TRUE
+	icon_state = "tcaf_dress_beret"
+	item_state = "tcaf_dress_beret"
+
+/obj/item/clothing/head/beret/legion/tcaf/tcaf_field
+	name = "TCAF field beret"
```
Comment (Generalcamo) [inline code/modules/clothing/head/berets.dm:204]:
```suggestion
	name = "\improper TCAF field beret"
```

File: code/modules/clothing/under/miscellaneous.dm [hunk 5]
```diff
@@ -540,6 +540,15 @@
 	item_state = "tauceti_pilot"
 	worn_state = "tauceti_pilot"
 
+/obj/item/clothing/under/legion/tcaf
+	name = "Tau Ceti Armed Forces armsman uniform"
```
Comment (Generalcamo) [inline code/modules/clothing/under/miscellaneous.dm:544]:
```suggestion
	name = "\improper Tau Ceti Armed Forces armsman uniform"
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/modular_armor.dm [hunk 6]
```diff
@@ -166,7 +180,7 @@
 	armor = list(
 		melee = ARMOR_MELEE_MINOR,
 		bullet = ARMOR_BALLISTIC_MINOR,
-		laser = ARMOR_LASER_MAJOR,
+		laser = ARMOR_LASER_RIFLE,
```
Comment (Generalcamo) [inline code/modules/clothing/suits/modular_armor.dm:183]: Do you have a reason behind this change? This isn't documented.

File: code/modules/clothing/suits/modular_armor.dm [hunk 7]
```diff
@@ -180,7 +194,7 @@
 	armor = list(
 		melee = ARMOR_MELEE_MAJOR,
 		bullet = ARMOR_BALLISTIC_MAJOR,
-		laser = ARMOR_LASER_RIFLE,
+		laser = ARMOR_LASER_MEDIUM,
```
Comment (Generalcamo) [inline code/modules/clothing/suits/modular_armor.dm:197]: Ditto

File: code/modules/clothing/suits/modular_armor.dm [hunk 8]
```diff
@@ -194,7 +208,7 @@
...
```
Comment (Generalcamo) [inline code/modules/clothing/suits/modular_armor.dm:211]: Ditto

File: code/modules/clothing/under/accessories/armor.dm [hunk 9]
```diff
@@ -39,7 +39,7 @@
 	armor = list(
 		melee = ARMOR_MELEE_MINOR,
 		bullet = ARMOR_BALLISTIC_MINOR,
-		laser = ARMOR_LASER_MAJOR,
+		laser = ARMOR_LASER_RIFLE,
```
Comment (Generalcamo) [inline code/modules/clothing/under/accessories/armor.dm:42]: Ditto

File: code/modules/clothing/under/accessories/armor.dm [hunk 10]
```diff
@@ -153,7 +168,7 @@
...
```
Comment (Generalcamo) [inline code/modules/clothing/under/accessories/armor.dm:171]: Ditto

File: code/modules/clothing/suits/modular_armor.dm (hunk 6 above)
Comment (NobleRow) [reply code/modules/clothing/suits/modular_armor.dm:183]: I honestly think I changed it out of curiosity and forgot to change it back. That was my first endeavor in to coding late at night and was just toying around with things before I got actual help from a server member today.

File: code/modules/clothing/suits/modular_armor.dm [hunk 11]
```diff
@@ -207,6 +221,39 @@
 	item_state = "plate_blue"
 	slowdown = 0 // the SCC is hacking
 
+/obj/item/clothing/accessory/armor_plate/tcaf
+	name = "tcaf legionnaire carapace"
```
Comment (Greenjoe12345) [inline code/modules/clothing/suits/modular_armor.dm:225]: in some places you've got tcaf and in others TCAF, wonder if it'd be better to have them consistant

Comment (NobleRow) [reply code/modules/clothing/suits/modular_armor.dm:225]: That's primarily based off of how the TCFL items were labeled in their respective codes. Sometimes the names had all caps, sometimes it was all lowercase, so I just matched what they had incase there was some weird code reason I didn't understand. It wouldn't be too hard though to make them all lowercase, but if I understand correctly, we'll need to undo the changes made earlier to make them improper, and those edits were made to both TCFL and TCAF gear.

**Comment by CyberSpyXD** [comment]
You will likely need to put in different sprite sheets for some of the aliens. Tajara, Unathi and IPCs in particular as their heads are different enough that usually the side sprites need adjusting, otherwise things look off. Check out "legion/helmet" for an example of that in the code.

Of course, save that for later when you know this is going to be put in though. Not sure if any spriters or lore will weigh in on what needs changing/adjusting/removal.

**Comment by NobleRow** [comment]
> You will likely need to put in different sprite sheets for some of the aliens. Tajara, Unathi and IPCs in particular as their heads are different enough that usually the side sprites need adjusting, otherwise things look off. Check out "legion/helmet" for an example of that in the code.
> 
> Of course, save that for later when you know this is going to be put in though. Not sure if any spriters or lore will weigh in on what needs changing/adjusting/removal.

I do plan to get to the other races, I just wanted to push a good foundation first to make sure I was doing everything right before I got in to the more advanced topics. The next stage is going to be working on voidsuits, more uniforms, and clothing for the other races that aren't already compatible. I do have an open-faced and glasses variant of the helmets that is more xeno-race friendly. Thank you though for the comment!

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 12]
```diff
@@ -263,6 +263,13 @@
 	item_state = "legion_bag"
 	empty_delay = 0.8 SECOND
 
+/obj/item/storage/backpack/tcaf
+	name = "\improper TCAF Carapace Backpack"
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/storage/backpack.dm:267]:
```suggestion
	name = "\improper TCAF carapace backpack"
```

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 13]
```diff
@@ -284,6 +284,16 @@
 	build_from_parts = TRUE
 	worn_overlay = "over"
 
+/obj/item/clothing/gloves/tcaf
+	name = "armsman gloves"
```
Comment (SleepyGemmy) [inline code/modules/clothing/gloves/miscellaneous.dm:288]:
```suggestion
	name = "\improper TCAF armsman gloves"
```

File: code/modules/clothing/suits/modular_armor.dm (hunk 11 above)
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:225]:
```suggestion
	name = "\improper TCAF legionnaire carapace"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 14]
```diff
@@ -207,6 +221,39 @@
...
+		melee = ARMOR_MELEE_MAJOR,
+		bullet = ARMOR_BALLISTIC_MAJOR,
+		laser = ARMOR_LASER_RIFLE,
+		energy = ARMOR_ENERGY_SMALL,
+		bomb = ARMOR_BOMB_PADDED,
+	)
+	slowdown = 0 // inherited the hacking from the scc
+
+/obj/item/clothing/accessory/armor_plate/tcaf/tcaf_light
+	name = "tcaf legionnaire light carapace"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:241]:
```suggestion
	name = "\improper TCAF legionnaire light carapace"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 15]
```diff
@@ -302,6 +349,32 @@
 		bomb = ARMOR_BOMB_PADDED,
 	)
 
+/obj/item/clothing/head/helmet/tcaf
+	name = "tcaf faceplate legionnaire helmet"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:353]:
```suggestion
	name = "\improper TCAF legionnaire faceplate helmet"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 16]
```diff
@@ -302,6 +349,32 @@
...
+	armor = list(
+		melee = ARMOR_MELEE_MAJOR,
+		bullet = ARMOR_BALLISTIC_MAJOR,
+		laser = ARMOR_LASER_MEDIUM,
+		energy = ARMOR_ENERGY_SMALL,
+		bomb = ARMOR_BOMB_PADDED,
+	)
+
+/obj/item/clothing/head/helmet/tcaf/tcaf_novisor
+	name = "tcaf legionnaire helmet"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:368]:
```suggestion
	name = "\improper TCAF legionnaire helmet"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 17]
```diff
@@ -302,6 +349,32 @@
...
+	desc = "A carapace helmet in the traditional colors of the Tau Ceti Armed Forces."
+	icon_state = "tcaf_helm_novisor"
+	item_state = "tcaf_helm_novisor"
+
+/obj/item/clothing/head/helmet/tcaf/tcaf_visor
+	name = "tcaf visored legionnaire helmet"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:374]:
```suggestion
	name = "\improper TCAF legionnaire visored helmet"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 18]
```diff
@@ -560,3 +633,23 @@
 	insignia of a well known unathi drapers' guild stitched on the back."
 	icon_state = "flagpatch_hegemony"
 	item_state = "flagpatch_hegemony"
+
+/obj/item/clothing/accessory/tcaf_prefect_pauldron
+	name = "tcaf prefect pauldron"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:638]:
```suggestion
	name = "\improper TCAF prefect pauldron"
```

File: code/modules/clothing/suits/modular_armor.dm [hunk 19]
```diff
@@ -560,3 +633,23 @@
...
+	desc = "A bright red hard pauldron to indicate the wearer has the rank of Prefect in the Tau Ceti Armed Forces."
+	icon = 'icons/clothing/kit/modular_armor.dmi'
+	icon_state = "tcaf_prefect_pauldron"
+	item_state = "tcaf_prefect_pauldron"
+	contained_sprite = TRUE
+	slot = ACCESSORY_SLOT_GENERIC
+	flippable = FALSE
+
+/obj/item/clothing/accessory/tcaf_senior_legion_pauldron
+	name = "tcaf senior legionnaire pauldron"
```
Comment (SleepyGemmy) [inline code/modules/clothing/suits/modular_armor.dm:648]:
```suggestion
	name = "\improper TCAF senior legionnaire pauldron"
```

File: code/modules/clothing/under/accessories/armor.dm [hunk 20]
```diff
@@ -113,6 +113,21 @@
 		bomb = ARMOR_BOMB_PADDED,
 	)
 
+/obj/item/clothing/accessory/leg_guard/tcaf
+	name = "tcaf legionnaire leg carapace"
```
Comment (SleepyGemmy) [inline code/modules/clothing/under/accessories/armor.dm:117]:
```suggestion
	name = "\improper TCAF legionnaire leg carapace"
```

File: code/modules/clothing/under/accessories/armor.dm [hunk 21]
```diff
@@ -239,3 +254,23 @@
 		energy = ARMOR_ENERGY_SMALL,
 		bomb = ARMOR_BOMB_PADDED,
 	)
+
+/obj/item/clothing/accessory/arm_guard/tcaf
+	name = "tcaf carapace arm guards"
```
Comment (SleepyGemmy) [inline code/modules/clothing/under/accessories/armor.dm:259]:
```suggestion
	name = "\improper TCAF carapace arm guards"
```

File: code/modules/clothing/under/accessories/armor.dm [hunk 22]
```diff
@@ -239,3 +254,23 @@
...
+	armor = list(
+		melee = ARMOR_MELEE_MAJOR,
+		bullet = ARMOR_BALLISTIC_MAJOR,
+		laser = ARMOR_LASER_MEDIUM,
+		energy = ARMOR_ENERGY_SMALL,
+		bomb = ARMOR_BOMB_PADDED,
+	)
+
+/obj/item/clothing/accessory/arm_guard/tcaf/tcaf_stripe
+	name = "striped tcaf carapace arm guards"
```
Comment (SleepyGemmy) [inline code/modules/clothing/under/accessories/armor.dm:274]:
```suggestion
	name = "striped TCAF carapace arm guards"
```

File: code/modules/clothing/under/miscellaneous.dm [hunk 23]
```diff
@@ -540,6 +540,15 @@
 	item_state = "tauceti_pilot"
 	worn_state = "tauceti_pilot"
 
+/obj/item/clothing/under/legion/tcaf
+	name = "\improper Tau Ceti Armed Forces armsman uniform"
```
Comment (SleepyGemmy) [inline code/modules/clothing/under/miscellaneous.dm:544]:
```suggestion
	name = "\improper TCAF armsman uniform"
```

File: html/changelogs/tcaf-additions.yml [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Added TCAF Uniform, Modular Armor, Backpack, Gloves, Berets and Pouches."
```
Comment (SleepyGemmy) [inline html/changelogs/tcaf-additions.yml:41]:
```suggestion
  - rscadd: "Added a TCAF uniform, modular armor, backpack, gloves, berets, and pouches."
```
american spelling.
```suggestion
  - rscadd: "Added a TCAF uniform, modular armour, backpack, gloves, berets, and pouches."
```
english spelling.
(pick either.)

**Comment by TheDanseMacabre** [comment]
The armor values for these are absolutely excessive. There's no reason for this armor to be this strong. It would be best if it had the same armor values as heavy armor. Right now this armor makes the wearer well-insulated against all attacks, and that precludes it being balanced for normal place. I see no reason why the TCAF's armor should be superior to any other human fighting force in the spur.

**Comment by NobleRow** [comment]
> The armor values for these are absolutely excessive. There's no reason for this armor to be this strong. It would be best if it had the same armor values as heavy armor. Right now this armor makes the wearer well-insulated against all attacks, and that precludes it being balanced for normal place. I see no reason why the TCAF's armor should be superior to any other human fighting force in the spur.

The armor is the same as heavy armor 1:1

Heavy armor is:
/obj/item/clothing/accessory/armor_plate/heavy
	name = "heavy armor plate"
	desc = "A heavy and menacing armor plate. Tan armor plates went out of style centuries ago!"
	icon_state = "plate_heavy"
	item_state = "plate_heavy"
	armor = list(
		melee = ARMOR_MELEE_MAJOR,
		bullet = ARMOR_BALLISTIC_MAJOR,
		laser = ARMOR_LASER_RIFLE,
		energy = ARMOR_ENERGY_SMALL,
		bomb = ARMOR_BOMB_PADDED,
	)

And the TCAF armor is:
/obj/item/clothing/accessory/armor_plate/tcaf
	name = "\improper TCAF legionnaire carapace"
	desc = "The blue carapace of the Tau Ceti Armed Forces. Polished and proud for Miranda Trasen's favorite soldiers."
	icon = 'icons/clothing/kit/modular_armor.dmi'
	icon_state = "tcaf_plate"
	item_state = "tcaf_plate"
	contained_sprite = TRUE
	armor = list(
		melee = ARMOR_MELEE_MAJOR,
		bullet = ARMOR_BALLISTIC_MAJOR,
		laser = ARMOR_LASER_RIFLE,
		energy = ARMOR_ENERGY_SMALL,
		bomb = ARMOR_BOMB_PADDED,
	)

I just copied the armor over and changed the names and the icon states.
There was recently a Laser Nerf that increased the armor for a lot of things against lasers though, and I adjusted the TCAF armor to match.
I never intended to make it more powerful than anything that exists and was just told to copy over heavy armor or military armor and switch it around. If there is anything that isn't the same do let me know so I can fix it!

**Comment by TheDanseMacabre** [comment]
> > --snip--

I must've mis-remembered the heavy armor values. That's my bad. Nevermind me.

**Review by SleepyGemmy — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**